eth/filters/api.go

```diff
@@ -556,13 +561,16 @@ func newFilterId() (string, error) {
 	return "0x" + hex.EncodeToString(subid[:]), nil
 }
 
-// returnLogs is a helper that will return an empty logs array case the given logs is nil, otherwise is will return the
-// given logs. The RPC interfaces defines that always an array is returned.
-func returnLogs(logs vm.Logs) vm.Logs {
-	if logs == nil {
-		return vm.Logs{}
+// toRPCLogs is a helper that will convert a vm.Logs array to an structure which
+// can hold additonal information about the logs such as whether it was deleted.
+// Additonally when nil is given it will by default instead create an empty slice
+// instead. This is required by the RPC specification.
+func toRPCLogs(logs vm.Logs, removed bool) []vmlog {
+	convertedLogs := make([]vmlog, len(logs))
+	for i, log := range logs {
+		convertedLogs[i] = vmlog{log, removed}
```

Comment:
I would abstain from using these shortened initializers (i.e. no field name, only the values) as it's brittle to future modifications (both type changes as well as order changes; in addition adding a new field breaks all code, even if the new fields' default value is to be used). In this specific instance I guess we could get away with in, but generally it's ill advised from the community.

Also good to know that on Google App Engine these result in compile errors specifically for the reason that old code can break.